source/agora/utils/TxBuilder.d

```diff
@@ -253,6 +253,16 @@ public struct TxBuilder
         return this;
     }
 
+    /// Sign with a given key and append ubytes if given
+    public static Unlock signWithSpecificKey (KeyPair key, ubyte[] append = null) (in Transaction tx, in OutputRef out_ref)
+        @safe nothrow
+    {
+        auto pair = SigPair(key.sign(tx.getChallenge()), SigHash.All);
+        if (append)
+            return Unlock([ubyte(65)] ~ pair[] ~ append);
+        return Unlock([ubyte(65)] ~ pair[].dup);
```

Comment:
Isn't the first byte a `length` ? Hence setting `append` to non-empty would not be valid ?

Reply:
```D
/// Contains a data tuple or a set of push opcodes
public struct Unlock
{
    /// May be: <signature>, <signature, key>, <push opcodes>
    public const(ubyte)[] bytes;

    ///
    public void toString (scope void delegate (scope const(char)[]) @safe sink)
        const @safe
    {
        sink("0x");
        UbyteHexString(this.bytes).toString(sink);
    }

    /// The size of the Unlock object
    public ulong sizeInBytes () const nothrow pure @safe @nogc
    {
        return this.bytes.length * this.bytes[0].sizeof;
    }
}
```

I don't understand why here we have the length preceding these `Unlock` when signing.
In other places it is not:
```D
/*******************************************************************************

    Generates a LockType.Key unlock script.

    Params:
        sig = the signature that will be embedded in the script

    Returns:
        the unlock script

*******************************************************************************/

public Unlock genKeyUnlock (in Signature sig) pure nothrow @safe
{
    // assume SigHash.All
    return genKeyUnlock(SigPair(sig));
}

/// Ditto
public Unlock genKeyUnlock (in SigPair sig) pure nothrow @safe
{
    // must dupe because it's a value on the stack..
    return Unlock(sig[].dup);
}
```

Reply:
If I change the first byte to `ubyte(64)` when there is no append the tests that use this fail.

Reply:
@omerfirmak is the meaning of the first byte the `length`?

Reply:
Yep it is and it is not needed when type of the Lock is `LockType.Key` or `LockType.KeyHash`. That is why you only see it in some places.